library/Ivoz/Provider/Application/Service/User/SyncFromCsv.php

```diff
@@ -70,52 +70,65 @@ public function execute(CompanyInterface $company, string $csv)
                     (int) $company->getId(),
                     ...$userArgs
                 );
-                $entities = [$user];
 
-                $notEmptyTerminalArgs = count(array_filter($terminalArgs)) > 0;
+                $terminal = null;
+                $notEmptyTerminalArgs = count($terminalArgs) === 4;
                 if ($notEmptyTerminalArgs) {
                     $terminal = $this->terminalFactory->fromMassProvisioningCsv(
                         (int) $company->getId(),
                         ...$terminalArgs
                     );
-
-                    $entities[] = $terminal;
+                    $this->entityTools->persist($terminal);
                 }
 
+                $extension = null;
                 if ($extensionArg) {
                     $extension = $this->extensionFactory->fromMassProvisioningCsv(
                         (int) $company->getId(),
                         $extensionArg,
-                        $user
+                        null
                     );
-
-                    $entities[] = $extension;
+                    $this->entityTools->persist($extension);
                 }
 
-                $notEmptyDdiArgs = count(array_filter($outboundDdiArgs)) > 0;
+                $ddi = null;
+                $notEmptyDdiArgs = count(array_filter($outboundDdiArgs, fn(mixed $value) => $value !== '' && $value !== null)) > 0;
                 if ($notEmptyDdiArgs) {
                     $ddi = $this->ddiFactory->fromMassProvisioningCsv(
                         $company,
                         ...$outboundDdiArgs
                     );
```

Comment:
It took me a while to understand what was happening here and what the intention was.
The  factory method is specting four arguments, so lets build a version with zero surprises:

```suggestion
                $outboundDdiArgs = array_pad($outboundDdiArgs, 2, '');
                $ddiArgsAreFilled = $ !(empty($outboundDdiArgs[0]) || empty($outboundDdiArgs[1]));
                
                if ($ddiArgsAreFilled) {
                    [$ddiNumber, $ddiProviderName] = $outboundDdiArgs;
                    $ddi = $this->ddiFactory->fromMassProvisioningCsv(
                    $company,
                    $ddiNumber,
                    $ddiProviderName,
                );
```

1) We make sure there are at least 2 args.
2) Then, check those are not empty.
3) We call the method with explicit arguments.